logprep/connector/s3/output.py

```diff
@@ -327,3 +262,77 @@ def store_failed(self, error_message: str, document_received: dict, document_pro
         }
         self._add_to_backlog(error_document, self._config.error_prefix)
         self._write_to_s3_resource()
+
+    def _add_dates(self, prefix):
+        date_format_matches = self._replace_pattern.findall(prefix)
+        if date_format_matches:
+            now = TimeParser.now()
+            for date_format_match in date_format_matches:
+                formatted_date = now.strftime(date_format_match[2:-1])
+                prefix = re.sub(date_format_match, formatted_date, prefix)
+        return prefix
+
+    @Metric.measure_time()
+    def _write_to_s3_resource(self):
+        """Writes a document into s3 bucket using given prefix."""
+        if self._backlog_size >= self._config.message_backlog_size:
+            self._write_backlog()
+
+    def _add_to_backlog(self, document: dict, prefix: str):
```

Comment:
please add type hint for return value
```suggestion
    def _add_to_backlog(self, document: dict, prefix: str) -> None:
```